src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -84,11 +90,73 @@ public static void ToCharsBuffer(byte value, Span<char> buffer, int startingInde
             buffer[startingIndex] = (char)(packedResult >> 8);
         }
 
+#if SYSTEM_PRIVATE_CORELIB
+    private static void EncodeToUtf16_Ssse3(ReadOnlySpan<byte> bytes, Span<char> chars, Casing casing)
+    {
+        int pos = 0;
+
+        Vector128<byte> shuffleMask = Vector128.Create(
+            0xFF, 0xFF, 0, 0xFF, 0xFF, 0xFF, 1, 0xFF,
+            0xFF, 0xFF, 2, 0xFF, 0xFF, 0xFF, 3, 0xFF);
+
+        Vector128<byte> asciiTable = (casing == Casing.Upper) ?
+            Vector128.Create((byte)'0', (byte)'1', (byte)'2', (byte)'3',
+                             (byte)'4', (byte)'5', (byte)'6', (byte)'7',
+                             (byte)'8', (byte)'9', (byte)'A', (byte)'B',
+                             (byte)'C', (byte)'D', (byte)'E', (byte)'F') :
+            Vector128.Create((byte)'0', (byte)'1', (byte)'2', (byte)'3',
+                             (byte)'4', (byte)'5', (byte)'6', (byte)'7',
+                             (byte)'8', (byte)'9', (byte)'a', (byte)'b',
+                             (byte)'c', (byte)'d', (byte)'e', (byte)'f');
+
+        for (; pos < bytes.Length - 3; pos += 4)
+        {
+            // Read 32bits from "bytes" span at "pos" offset
+            uint block = Unsafe.ReadUnaligned<uint>(
+                ref Unsafe.Add(ref MemoryMarshal.GetReference(bytes), pos));
```

Comment:
Is there a difference in code-gen if the `ref MemoryMarshal.GetReference` is hoisted outside the loop?

Reply:
Just checked - no effect on final codegen.